### PR TITLE
lighting: increase pd client retry count on error during initialization

### PR DESCRIPTION
**Pull request by tangenta**
### What problem does this PR solve?

Issue Number: close #xxx

Problem Summary:

```go
func (c *pdServiceDiscovery) initRetry(f func() error) error {
	var err error
	ticker := time.NewTicker(time.Second)
	defer ticker.Stop()
	for i := 0; i < c.option.maxRetryTimes; i++ {
		if err = f(); err == nil {
			return nil
		}
		select {
		case <-c.ctx.Done():
			return err
		case <-ticker.C:
		}
	}
	return errors.WithStack(err)
}
```

The retry times of `initRetry` is too small:

```
[2024/03/11 07:38:42.674 +00:00] [WARN] [pd_service_discovery.go:861] ["[pd] failed to get cluster id"] [url=http://tidb-pd-0.tidb-pd-peer.sqllogic-tps-7500342-1-143.svc:2379] [error="[PD:client:ErrClientGetMember]error:rpc error: code = Unavailable desc = connection error: desc = \"transport: Error while dialing: dial tcp: lookup tidb-pd-0.tidb-pd-peer.sqllogic-tps-7500342-1-143.svc: i/o timeout\" target:tidb-pd-0.tidb-pd-peer.sqllogic-tps-7500342-1-143.svc:2379 status:TRANSIENT_FAILURE: error:rpc error: code = Unavailable desc = connection error: desc = \"transport: Error while dialing: dial tcp: lookup tidb-pd-0.tidb-pd-peer.sqllogic-tps-7500342-1-143.svc: i/o timeout\" target:tidb-pd-0.tidb-pd-peer.sqllogic-tps-7500342-1-143.svc:2379 status:TRANSIENT_FAILURE"]
[2024/03/11 07:38:42.674 +00:00] [WARN] [pd_service_discovery.go:861] ["[pd] failed to get cluster id"] [url=http://tidb-pd-0.tidb-pd-peer.sqllogic-tps-7500342-1-143.svc:2379] [error="[PD:client:ErrClientGetMember]error:rpc error: code = Unavailable desc = connection error: desc = \"transport: Error while dialing: dial tcp: lookup tidb-pd-0.tidb-pd-peer.sqllogic-tps-7500342-1-143.svc: i/o timeout\" target:tidb-pd-0.tidb-pd-peer.sqllogic-tps-7500342-1-143.svc:2379 status:TRANSIENT_FAILURE: error:rpc error: code = Unavailable desc = connection error: desc = \"transport: Error while dialing: dial tcp: lookup tidb-pd-0.tidb-pd-peer.sqllogic-tps-7500342-1-143.svc: i/o timeout\" target:tidb-pd-0.tidb-pd-peer.sqllogic-tps-7500342-1-143.svc:2379 status:TRANSIENT_FAILURE"]
[2024/03/11 07:38:43.673 +00:00] [WARN] [pd_service_discovery.go:861] ["[pd] failed to get cluster id"] [url=http://tidb-pd-0.tidb-pd-peer.sqllogic-tps-7500342-1-143.svc:2379] [error="[PD:client:ErrClientGetMember]error:rpc error: code = Unavailable desc = connection error: desc = \"transport: Error while dialing: dial tcp: lookup tidb-pd-0.tidb-pd-peer.sqllogic-tps-7500342-1-143.svc: i/o timeout\" target:tidb-pd-0.tidb-pd-peer.sqllogic-tps-7500342-1-143.svc:2379 status:TRANSIENT_FAILURE: error:rpc error: code = Unavailable desc = connection error: desc = \"transport: Error while dialing: dial tcp: lookup tidb-pd-0.tidb-pd-peer.sqllogic-tps-7500342-1-143.svc: i/o timeout\" target:tidb-pd-0.tidb-pd-peer.sqllogic-tps-7500342-1-143.svc:2379 status:TRANSIENT_FAILURE"]
[2024/03/11 07:38:44.673 +00:00] [INFO] [pd_service_discovery.go:664] ["[pd] close pd service discovery client"]
[2024/03/11 07:38:44.673 +00:00] [ERROR] [backend_mgr.go:142] ["build ingest backend failed"] [category=ddl-ingest] ["job ID"=5419] [error="[Lightning:PD:ErrCreatePDClient]create pd client error: [pd] failed to get cluster id"]
[2024/03/11 07:38:44.678 +00:00] [WARN] [ddl_worker.go:1132] ["run DDL job error"] [worker="worker 4, tp add index"] [category=ddl] [jobID=5419] [conn=1098908168] [error="[Lightning:PD:ErrCreatePDClient]create pd client error: [pd] failed to get cluster id"]
...
[2024/03/11 07:38:46.041 +00:00] [INFO] [pd_service_discovery.go:1043] ["[pd] switch leader"] [new-leader=http://tidb-pd-0.tidb-pd-peer.sqllogic-tps-7500342-1-143.svc:2379] [old-leader=]
[2024/03/11 07:38:46.041 +00:00] [INFO] [pd_service_discovery.go:525] ["[pd] init cluster id"] [cluster-id=7344998754855476514]
```

We can see PD cluster ID is initialized after two seconds.

### What changed and how does it work?

This PR removes pd client retry count on error during initialization, and let it use default value `100`:

```
const (
	defaultPDTimeout                             = 3 * time.Second
	maxInitClusterRetries                        = 100                  // <-----
	defaultMaxTSOBatchWaitInterval time.Duration = 0
	defaultEnableTSOFollowerProxy                = false
	defaultEnableFollowerHandle                  = false
)
```

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by ywqzzy — APPROVED** [review]

File: br/pkg/lightning/backend/local/local.go [hunk 1]
```diff
@@ -517,7 +517,7 @@ func NewBackend(
 		// If the time too short, we may scatter a region many times, because
 		// the interface `ScatterRegions` may time out.
 		pd.WithCustomTimeoutOption(60*time.Second),
-		pd.WithMaxErrorRetry(3),
+		pd.WithMaxErrorRetry(100),
```
Comment (lance6716) [inline br/pkg/lightning/backend/local/local.go:520]: 100 is the default value, I prefer we search and remove all `WithMaxErrorRetry`

Comment (tangenta) [reply br/pkg/lightning/backend/local/local.go:520]: Removing all of these will also affect the import by lightning process. If there is really something wrong with PD (like #48209), the error will be reported after at least 100 seconds, is that expected?

**Review by lance6716 — APPROVED** [review]

**Review by joccau — APPROVED** [review]
LGTM

**Review by YuJuncen — APPROVED** [review]

**Pull request merged**